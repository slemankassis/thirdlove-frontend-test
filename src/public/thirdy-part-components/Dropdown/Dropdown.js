const React = require('react');


class Dropdown extends React.Component {
  render() {
    return (
      1
    );
  }
}

module.exports = Dropdown;
