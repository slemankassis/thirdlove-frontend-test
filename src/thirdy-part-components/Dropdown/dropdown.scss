  .css-yk16xz-control, .css-2b097c-container {
    width: 130px;
  }
