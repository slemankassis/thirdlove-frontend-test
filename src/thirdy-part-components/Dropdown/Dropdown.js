import React from 'react';
import PropTypes from 'prop-types';
import './dropdown.scss';

class Dropdown extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);

    this.state = {
      selected: this.props.selected,
    };
  }

  handleChange(e) {
    const { value } = e.target;
    if (value.value !== this.props.selected.value) {
      this.setState(() => ({
        selected: value,
      }));
      this.props.onChange(value);
    }
  }

  render() {
    const { options, label } = this.props;

    return (
      <div className="dropdown">
        <label htmlFor={this.state.selected.value}>
          {label}
          <select onChange={this.handleChange} value={this.props.selected}>
            {(options).map((value) => (
              <option key={value} value={value} checked={value === this.props.selected}>
                {value}
              </option>
            ))}
          </select>
        </label>
      </div>
    );
  }
}

Dropdown.propTypes = {
  selected: PropTypes.shape({
    value: PropTypes.string,
    label: PropTypes.string,
  }),
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
  label: PropTypes.string,
};

Dropdown.defaultProps = {
  selected: undefined,
  label: '',
};
export default Dropdown;
