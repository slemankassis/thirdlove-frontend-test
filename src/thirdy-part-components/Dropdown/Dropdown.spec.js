import React from 'react';
import Dropdown from './Dropdown';

describe('Dropdown', () => {
  it('renders snapshot with basicProps', () => {
    const basicProps = {
      selected: {
        value: '__VALUE__',
        label: '__LABEL__',
      },
      options: ['__OPTION_1__', '__OPTION_2__'],
      onChange: jest.fn(),
    };

    const component = shallow(<Dropdown {...basicProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('renders snapshot with fullProps', () => {
    const fullProps = {
      selected: {
        value: '__VALUE__',
        label: '__LABEL__',
      },
      options: ['__OPTION_1__', '__OPTION_2__'],
      onChange: jest.fn(),
      label: '__LABEL__',
    };

    const component = shallow(<Dropdown {...fullProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
