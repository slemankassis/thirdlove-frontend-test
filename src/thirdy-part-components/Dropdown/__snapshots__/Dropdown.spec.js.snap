// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown snaps renders snapshot with basicProps 1`] = `
"<div className='dropdown'>
  <label htmlFor='__VALUE__'>
    <StateManager value={{...}} onChange={[Function: bound handleChange]} options={{...}} defaultInputValue='' defaultMenuIsOpen={false} defaultValue={{...}} />
  </label>
</div>"
`;

exports[`Dropdown snaps renders snapshot with fullProps 1`] = `
"<div className='dropdown'>
  <label htmlFor='__VALUE__'>
    __LABEL__
    <StateManager value={{...}} onChange={[Function: bound handleChange]} options={{...}} defaultInputValue='' defaultMenuIsOpen={false} defaultValue={{...}} />
  </label>
</div>"
`;
